### PR TITLE
Azure deployment templates and docs

### PR DESCRIPTION
**Pull request by marrobi**
This PR includes the templates and steps required to deploy OK.py to Azure. This is the last PR I'm expecting from the work in May other than bug fixes. Welcome feedback, including directory structure etc.

@liliankasem  @cicorias @martinpeck @c-w can you have a quick check over the "integration guide". If  Would be great if you can test deployment to your Azure accounts. 

One issue that has a workaround is #1279 . Need to regenerate key and redeploy with identical settings if the app doesn't come up. Kudu logs will show issue parsing Redis URL

@lawrencegripper do you mind casting your eyes over the templates. I will likely do some further work at a later point in time, but what is there works which is what matters at the moment.

@colinschoen @Sumukh if you want/need Azure credit to test this let me know. I've just learnt there is an article going out from Imperial about using Ok.py early next week I believe so would be great if have it merged by then!

Thanks for the continued support.

### CODE REVIEW

**Comment by colinschoen** [comment]
I'll take a look today.

File: azure/paas/aci-create-container.deploy.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        "restartPolicy": "Never",
+        "containers": [
+          {
+            "name": "[concat('create-storage-container', uniquestring(resourceGroup().id))]",
+            "properties": {
+              "image": "microsoft/azure-cli:2.0.31",
+              "command": [
+                "sh",
+                "-c",
+                "az storage container create --account-name=$STORAGE_ACCOUNT_NAME --account-key=$STORAGE_KEY --name=$STORAGE_CONTAINER_NAME"
```
Comment (lawrencegripper) [inline azure/paas/aci-create-container.deploy.json]: This is a cool trick!

**Comment by c-w** [comment]
@colinschoen The CI is failing because wtforms "can't pickle dict_items objects" (likely linked to a transitive dependency update). I did a bit of digging and some people had similar problems claiming it's linked to code like `choices=some_dict.items()`. This construct indeed exists in quite a few of the files in the code (e.g. `BatchEnrollmentForm`, `EnrollmentExportForm`, 'VersionForm`, etc). For some reason I can't reproduce this locally, so I've pushed a tentative fix (dd3c11f) to Travis to see what happens.

**Comment by c-w** [comment]
That fix worked (see [build 378](https://travis-ci.org/icokpy/ok/builds/389707548)). Let me know if you'd prefer this as a separate pull request or if you're happy to merge the fix as part of this PR.

**Pull request merged**